contracts/Token/CustomToken.sol

```diff
@@ -0,0 +1,379 @@
+     */
+    function name() public view returns(string memory) { return _name; }
+
+    /**
+     * @dev returns count of tokens
+     * @return symbol of token
+     */
+    function symbol() public view returns(string memory) { return _symbol; }
+
+
+    /**
+     * @dev getting balance of tokens for {_user}
+     * @param _user address of user 
+     * @return balance
+     */
+    function balanceOf(
+        address _user
+    ) 
+        public 
+        view 
+        returns (uint256 balance) 
+    {
+        return balances[_user];
+    }
+
+    /**
+     * @dev add ballot project to list
+     * @param _project address of ballot project
+     * @return success
+     */
+    function addToProjects(
+        address _project
+    ) 
+        public
+        onlyOwner
+        returns (bool success)
+    {
+        require(_project != address(0), "Address must be non-empty");
+        require(!projectExists[_project], "Address already in list");
+        require(projects.length < 11, "Limit of projects are riched");
+        projects.push(_project);
+        projectExists[_project] = true;
+        projectsIndexes[_project] = projects.length - 1;
+        emit ProjectAdded(_project);
+        return true;
+    }
+
+    /**
+     * @dev removing projects from the list
+     * can called only by project, which will be removed
+     * @param _project address of ZeroOne project
+     */
+    function removeFromProjects(
+        address _project
+    ) 
+        public 
+        onlyZeroOne
+    {
+        uint index = projectsIndexes[_project];
+        address lastProjectInList = projects[projects.length - 1];
+
+        projectExists[_project] = false;
+        projectsIndexes[_project] = 0;
+
+        projects[index] = lastProjectInList;
+        projectsIndexes[lastProjectInList] = index;
+
+        projects.pop();
+        emit ProjectRemoved(_project);
+    }
+
+    /**
+     * @dev Transfers tokens from _sender to _recipient
+     * @param _sender address of token sender
+     * @param _recipient address of recipient of tokens
+     * @param _count count of tokens, which will be transferred
+     */
+    function transfer(
+        address _sender, 
+        address _recipient, 
+        uint256 _count
+    )
+        internal 
+    {
+        require(balances[_sender] >= _count, "Value must be less or equal user balance");
+        uint256 newSenderBalance = balances[_sender] - _count;
+        uint256 newRecipientBalance = balances[_recipient] + _count;
+
+        if (balances[_recipient] == 0) {
+            addHolder(_recipient);
+        }
+
+        balances[_sender] = newSenderBalance;
+        balances[_recipient] = newRecipientBalance;
+
+        if (newSenderBalance == 0) {
+            removeHolder(_sender);
+        }
+        emit Transfer(_sender, _recipient, _count);
+    }
+
+    /**
+     * @dev Function, which find ZeroOne projects, where user is voted
+     * and updating votes in this project
```

Comment:
finds ZeroOne projects, where user has voted and updates its votes in this project